neutone_sdk/parameter.py

```diff
@@ -36,26 +48,74 @@ def to_metadata_dict(self) -> Dict[str, str]:
         }
 
 
-class KnobNeutoneParameter(NeutoneParameter):
+class ContinuousNeutoneParameter(NeutoneParameter):
     """
-    Defines a knob Neutone Parameter that the user can use to control a model.
+    Defines a continuous Neutone Parameter that the user can use to control a model.
 
     The name and the description of the parameter will be shown as a tooltip
-    within the UI. `default_value` must be between 0 and 1 and will be used
-    as a default in the plugin when no presets are available.
+    within the UI.
+    `default_value` must be between 0 and 1 and will be used as a default in the plugin
+    when no presets are available.
+
+    The continuous parameter is considered to be a knob parameter.
     """
     def __init__(self,
                  name: str,
                  description: str,
                  default_value: float,
                  used: bool = True):
         super().__init__(name, description, used)
+        assert 0.0 <= default_value <= 1.0, \
+            "`default_value` for continuous params must be between 0 and 1"
+        self.default_value = default_value
+        self.type = NeutoneParameterType.CONTINUOUS
+
+    def to_metadata_dict(self) -> Dict[str, str]:
+        """Returns a string dictionary containing the metadata of the parameter."""
+        data = super().to_metadata_dict()
+        data["default_value"] = str(self.default_value)
+        return data
+
+
+class CategoricalNeutoneParameter(NeutoneParameter):
+    """
+    Defines a categorical Neutone Parameter that the user can use to control a model.
+
+    The name and the description of the parameter will be shown as a tooltip
+    within the UI.
+    `n_values` must be an int greater than or equal to 2 and less than or equal to 32.
+    `default_value` must be in the range [0, `n_values` - 1].
+    `labels` is a list of strings that will be used as the labels for the parameter.
+
+    The categorical parameter is considered to be a knob parameter.
+    """
+    def __init__(self,
+                 name: str,
+                 description: str,
+                 n_values: int,
+                 default_value: int,
+                 labels: Optional[List[str]] = None,
+                 used: bool = True):
+        super().__init__(name, description, used)
+        assert 2 <= n_values <= constants.MAX_N_CATEGORICAL_VALUES, \
+            "`n_values` for categorical params must between 2 and 20"
+        assert 0 <= default_value <= n_values - 1, \
+            "`default_value` for categorical params must be between 0 and `n_values`-1"
+        self.n_values = n_values
```

Comment:
Just an idea for this one, but wouldn't it be easier to just have to define the list of values that the parameter can take? (Which could be int/float/string etc)
Also set default_value=None by default which defaults to the first one in the list.

The idea is for the minimum things needed to define the parameter to be something like
```python
CategoricalNeutoneParameter(name="num_passes", description="", values=[1,2,3])
```
Not feeling too strongly about this, but I think the current API is a bit confusing.

Reply:
Yeah that's a good point, I think I was thinking that on the VST side we can only support ints from 0 to N but I don't see why that's the case. Will change it to this.

Reply:
Oh actually never mind, the reason I did it like this is because the categorical params are implemented behind the scenes using integer tensors in pytorch and needs to support param aggregation etc. It may still be possible to do this by adding some additional under the hood plumbing, will take a look at this.

Reply:
Turns out this is quite messy to do and get the typing to work out. I think we should stick with the current approach.